File: hackerrank.com/problem-solving/electronic_shop.scala
// https://www.hackerrank.com/challenges/electronics-shop/problem

import scala.io.StdIn._

object Solution {
  
  def getMoneySpent(keyboards: Array[Int], drives: Array[Int], b: Int): Int = {
    val prices = for (k <- keyboards; d <- drives) yield k + d
    val filtered = prices.filter(_ <= b)
    if (filtered.isEmpty) -1 else filtered.max
  }


  def main(args: Array[String]) {
    // val n = readInt
    val bnm = readLine.split(" ").map(_.trim.toInt)
    val keyboards = readLine.split(" ").map(_.trim.toInt)
    val drives = readLine.split(" ").map(_.trim.toInt)
    println(getMoneySpent(keyboards, drives, bnm(0)))
  }
}

File: hackerrank.com/problem-solving/electronic_shop.js
// /https://www.hackerrank.com/challenges/electronics-shop/problem

function getMoneySpent(keyboards, drives, b) {
  keyboards = keyboards.sort((a, b) => b - a);
  drives = drives.sort((a, b) => b - a);
  if (keyboards[keyboards.length -1] + drives[drives.length - 1] > b) return -1;
  if (keyboards[0] + drives[0] <= b) return keyboards[0] + drives[0];
  let maxVal = 0;
  for (let i = 0; i < keyboards.length; i++) {
    for (let j = 0; j < drives.length; j++) {
      const cost = keyboards[i] + drives[j]
      if (cost === b)
        return b
      else {
        if (cost < b) {
          maxVal = Math.max(maxVal, cost)
          break;
        }
      }
    }
  }
  return maxVal;
}

function main(inputData) {
  const lines = inputData.split("\n")
  const [b, n, m] = lines[0].split(' ').map(x => parseInt(x))
  const keyboards = lines[1].split(' ').map(x => parseInt(x))
  const drives = lines[2].split(' ').map(x => parseInt(x))
  console.log(getMoneySpent(keyboards, drives, b))
}

process.stdin.resume();
process.stdin.setEncoding("ascii");
let _input = '';
process.stdin.on("data", input => {
  _input += input;
});

process.stdin.on("end", () => {
 main(_input);
});